script.js

```diff
@@ -1,4 +1,443 @@
+        if (playGameCount == userNames.length - 1) {
+          let sortedScoreBoard = [...scoreBoard].sort(function (a, b) {
+            return b - a;
+          });
+          console.log(sortedScoreBoard + " :sorted scoreBoard");
+          // to find who is in the lead
+          return `<b>${
+            userNames[playGameCount]
+          }</b> rolled ${diceRolls}, resulting in a score of ${
+            tempScoreBoard[playGameCount]
+          }.<br><br><b>${
+            userNames[scoreBoard.indexOf(sortedScoreBoard[0])]
+          }</b> is currently in the lead. Please buck up the rest of you!<br><br> Click submit to continue playing! ${
+            ((playGameCount = 0), (diceRolls = []), (tempScoreBoard = []))
+          }`;
+        }
+
+        //increase playGameCount to exit this if condition
+        playGameCount += 1;
+
+        return `<b>${
+          userNames[playGameCount - 1]
+        }</b> rolled ${diceRolls}, resulting in a score of ${
+          tempScoreBoard[playGameCount - 1]
+        }.<br><br><b>${
+          userNames[playGameCount]
+        }</b> click submit to roll. ${(diceRolls = [])}`;
+      }
+
+      // Lowest game type
+      else if (gameType == LOWEST.toUpperCase()) {
+        ////sort diceRolls array ascending
+        let sortedDiceRolls = [...diceRolls].sort(function (a, b) {
+          return a - b;
+        });
+        console.log(sortedDiceRolls + " :sorted diceRolls");
+
+        //concatenate diceRolls array and push into scoreboard array
+        scoreBoard[playGameCount] += Number(sortedDiceRolls.join(""));
+        console.log(scoreBoard + " :scoreboard");
+
+        tempScoreBoard.push(sortedDiceRolls.join(``));
+
+        // for final player roll statement and leaderboard
+        if (playGameCount == userNames.length - 1) {
+          let sortedScoreBoard = [...scoreBoard].sort(function (a, b) {
+            return a - b;
+          });
+          console.log(sortedScoreBoard + " :sorted scoreBoard");
+          // to find who is in the lead
+          return `<b>${
+            userNames[playGameCount]
+          }</b> rolled ${diceRolls}, resulting in a score of ${
+            tempScoreBoard[playGameCount]
+          }.<br><br><b>${
+            userNames[scoreBoard.indexOf(sortedScoreBoard[0])]
+          }</b> is currently in the lead. Please buck up the rest of you!<br><br> Click submit to continue playing! ${
+            ((playGameCount = 0), (diceRolls = []), (tempScoreBoard = []))
+          }`;
+        }
+
+        //increase playGameCount to exit this if condition
+        playGameCount += 1;
+
+        return `<b>${
+          userNames[playGameCount - 1]
+        }</b> rolled ${diceRolls}, resulting in a score of ${
+          tempScoreBoard[playGameCount - 1]
+        }.<br><br><b>${
+          userNames[playGameCount]
+        }</b> click submit to roll. ${(diceRolls = [])}`;
+      }
+    }
+  }
+  //KNOCKOUT game mode
+  else if (gameMode == KNOCKOUT.toUpperCase()) {
```

Comment:
In this case, actually we could've just used 'KOCKOUT' (i.e. all caps) string in line 18. Then we wouldn't have to transform it to upper.

Reply:
Thank you for taking the time to review! really appreciate it!